src/xdocs/config_design.xml

```diff
@@ -1515,14 +1547,120 @@ class Test {
+
+ public final Set&lt;String&gt; mySet1 = new HashSet&lt;&gt;();  // violation, HashSet is mutable
+ public final ImmutableSet&lt;String&gt; mySet2;           // violation
+                                                     // immutable type not in config
+ public final ImmutableMap&lt;String, Object&gt; objects1; // violation
+                                                     // immutable type not in config
+
+ @com.annotation.CustomAnnotation
+ String annotatedString;
+
+ @CustomAnnotation
+ String shortCustomAnnotated;
+
+ @com.google.common.annotations.VisibleForTesting
+ public String testString = "";
+}
+        </source>
+        <p>
+          To configure the check so that it allows protected visible members:
+        </p>
+        <source>
+&lt;module name=&quot;VisibilityModifier&quot;&gt;
+  &lt;property name=&quot;protectedAllowed&quot; value=&quot;true&quot;/&gt;
+&lt;/module&gt;
+        </source>
+        <p>Example of allowed protected visible members:</p>
+        <source>
+public class MyClass {
+ private int myPrivateField1;
+
+ int field1;               // violation, must have a visibility modifier
 
+ protected String field2;
+
+ public int field3 = 42;  // violation, not static final, immutable,
+                          // nor matching the pattern
+ public long serialVersionUID = 1L;
+ public static final int field4 = 42;
+
+ public final int field5 = 42;          // violation, public immutable fields are not allowed
+ public final java.lang.String notes;   // violation, public immutable fields are not allowed
+
+ public final Set&lt;String&gt; mySet1 = new HashSet&lt;&gt;();  // violation, HashSet is mutable
+ public final ImmutableSet&lt;String&gt; mySet2;           // violation
+                                                     // immutable type not in config
+ public final ImmutableMap&lt;String, Object&gt; objects1; // violation
+                                                     // immutable type not in config
+
+ @com.annotation.CustomAnnotation
+ String annotatedString;      // violation, annotation is not configured
+
+ @CustomAnnotation
+ String shortCustomAnnotated; // violation, annotation is not configured
+
+ @com.google.common.annotations.VisibleForTesting
+ public String testString = "";
+}
+        </source>
         <p>
           To configure the check so that it allows no public members:
         </p>
         <source>
 &lt;module name=&quot;VisibilityModifier&quot;&gt;
   &lt;property name=&quot;publicMemberPattern&quot; value=&quot;^$&quot;/&gt;
 &lt;/module&gt;
+        </source>
+        <p>Example of not allowed public members:</p>
+        <source>
+public class MyClass {
+ private int myPrivateField1;
+
+ int field1;               // violation, must have a visibility modifier
+
+ protected String field2;  // violation, protected visibility is not allowed
+
+ public int field3 = 42;  // violation, not static final, immutable,
+                          // nor matching the pattern
```

Comment:
Comments are still not vertically aligned.

Reply:
Done.

Reply:
Resolved.